tools/MakeNiiFilesBIDSCompliant.pl

```diff
@@ -492,13 +497,11 @@ sub makeNIIAndHeader {
                 $headerVal    =   &fetchMincHeader($mincFileName,$headerNameMINC);
                 # Some headers need to be explicitely converted to floats in Perl
                 # so json_encode does not add the double quotation around them
-                if (($headerNameMINC eq 'acquisition:repetition_time') ||
-                    ($headerNameMINC eq 'acquisition:echo_time') ||
-                    ($headerNameMINC eq 'acquisition:inversion_time') ||
-                    ($headerNameMINC eq 'dicom_0x0018:el_0x1314')) {
-                        $headerVal *= 1;
-                }
-
+                my @convertToFloat = [
+                    'acquisition:repetition_time', 'acquisition:echo_time',
+                    'acquisition:inversion_time',  'dicom_0x0018:el_0x1314'
+                ];
+                $headerVal *= 1 if ($headerVal && $headerNameMINC ~~ @convertToFloat);
```

Comment:
for this one, it happens sometimes that one of them returns undef that is after converted into a float returning an error but continuing with a new $headerVal=0. So I check first that there $headerVal is defined and check if the header is one of the header that needs conversion to float

Reply:
this was in my TODO list in the PR description, I guess I can cross it now:)